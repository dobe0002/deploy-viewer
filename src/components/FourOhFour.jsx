import React from 'react';
// import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import { Alert } from 'reactstrap';

const FourOhFour = () => {
  return (
    <Alert color='danger'>
      Page not found. Please visit the
      <span> </span>
      <Link exact to='/deploys' className='alert-link'>
        Deploy Status page
      </Link>
    </Alert>
  );
};

export default FourOhFour;
