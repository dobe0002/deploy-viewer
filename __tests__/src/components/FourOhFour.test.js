/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import FourOhFour from '../../../src/components/FourOhFour';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('404 component tests', () => {
  test('Accessibility check', async () => {
    const wrapper = mount(
      <Router>
        <FourOhFour />
      </Router>
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
  test('Snapshot test', () => {
    const wrapper = mount(
      <Router>
        <FourOhFour />
      </Router>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
