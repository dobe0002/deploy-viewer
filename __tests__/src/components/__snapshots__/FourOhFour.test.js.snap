// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 component tests Snapshot test 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FourOhFour>
      <Alert
        closeAriaLabel="Close"
        color="danger"
        fade={true}
        isOpen={true}
        tag="div"
        transition={
          Object {
            "appear": true,
            "baseClass": "fade",
            "baseClassActive": "show",
            "enter": true,
            "exit": true,
            "in": true,
            "mountOnEnter": false,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 150,
            "unmountOnExit": true,
          }
        }
      >
        <Fade
          appear={true}
          baseClass="fade"
          baseClassActive="show"
          className="alert alert-danger"
          enter={true}
          exit={true}
          in={true}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          role="alert"
          tag="div"
          timeout={150}
          unmountOnExit={true}
        >
          <Transition
            appear={true}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={150}
            unmountOnExit={true}
          >
            <div
              className="alert alert-danger fade"
              role="alert"
            >
              Page not found. Please visit the
              <span>
                 
              </span>
              <Link
                className="alert-link"
                exact={true}
                to="/deploys"
              >
                <LinkAnchor
                  className="alert-link"
                  exact={true}
                  href="/deploys"
                  navigate={[Function]}
                >
                  <a
                    className="alert-link"
                    exact={true}
                    href="/deploys"
                    onClick={[Function]}
                  >
                    Deploy Status page
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </Transition>
        </Fade>
      </Alert>
    </FourOhFour>
  </Router>
</BrowserRouter>
`;
